feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/nft/search/NftSearchInteractor.kt

```diff
@@ -21,30 +23,36 @@ class NftSearchInteractor @Inject constructor(
 
     fun sendNftSearch(queryFlow: Flow<String>): Flow<Map<Chain, List<SendNftListItem>>> {
         return combine(getUserNfts(), queryFlow) { nfts, query ->
-            val supportedNfts = nfts
-                .filter { nftRepository.isNftTypeSupportedForSend(it.type) }
-
-            mapNftToNftDetails(supportedNfts)
-                .map(::mapNftDetailsToListItem)
-                .filterByQuery(query)
+            nfts.filterByQuery(query)
                 .groupBy { nftDetails -> nftDetails.chain }
         }
     }
 
-    private fun getUserNfts(): Flow<List<Nft>> {
+    private fun getUserNfts(): Flow<List<SendNftListItem>> {
         return accountRepository.selectedMetaAccountFlow()
             .flatMapLatest(nftRepository::allNftFlow)
+            .map { nfts ->
+                nfts.filter { nftRepository.isNftTypeSupportedForSend(it.type) }
+            }
+            .map { nfts ->
+                val nftLocals = nftRepository.getLocalNfts(nfts.map { it.identifier })
+                fullNftSyncIfNotSynced(nftLocals, nfts)
+                nftLocals
+                    .mapNotNull { nftRepository.nftDetails(it.identifier).firstOrNull() }
+                    .map(::mapNftDetailsToListItem)
+            }
```

Comment:
I have a few concerns here
1. Is that really necessary to request same nft list twice (`nftRepository::allNftFlow` and `nftRepository.getLocalNfts`)?
2. Seems like we're trying to sync every NFT in the list which might be an issue for users that have thouthands of NFTs. This is one of test-cases during QA verification btw. I suggest using same approach we use in nft list, where we only load those that are visible on the screen
3. We're still making a separate database request for each nft to create nftDetails

Reply:
I had similar questions :) 
We need sync every NFT because we need collection name for every NFT. I can get collection name only if `wholeDetailsLoaded = true`. `wholeDetailsLoaded = true` when there was full NFT sync

Reply:
Right, we need collection name, but on the same time, we only need it for those items that are already loaded. Meaning we can load it lazily - right after we're about to bind list item in RecyclerView. We did the similar thing in nft list

Reply:
Syncing all nfts on large accounts that way will most likely crash or freeze the app due to memory consuption, especcialy on lower-end devices so we should do things lazy as much as possible